### PR TITLE
purple-discord: init at 2018-04-10

### PR DESCRIPTION
**Pull request by S-NA**
###### Motivation for this change
Adds new pidgin plugin that brings Discord support. In addition, add myself to the maintainers list.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

@jtojnar mind reviewing this again? Sorry about the new pull request, did not know GitHub would close it, when I forced a push to squash my commits. Also thank you for telling me about `PKG_CONFIG_${package}_${variable}` it is awesome.

### CODE REVIEW

**Comment by jtojnar** [comment]
@GrahamcOfBorg build pidgin-discord

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-discord/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  };
+
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ pidgin json_glib ];
+
+  makeFlags = [
+    "DESTDIR=$(out)"
+  ];
+
+  PKG_CONFIG_PURPLE_PLUGINDIR = "/lib/pidgin";
```
Comment (jtojnar) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-discord/default.nix]: This should probably be `lib/purple-2`, see https://github.com/NixOS/nixpkgs/pull/38311.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -17278,6 +17278,8 @@ with pkgs;
     plugins = [];
   };
 
+  pidgin-discord = callPackage ../applications/networking/instant-messengers/pidgin-plugins/pidgin-discord { };
```
Comment (jtojnar) [inline pkgs/top-level/all-packages.nix]: Why are you naming it `pidgin-discord` when the repo is called `purple-discord`?

**Review by jtojnar — CHANGES_REQUESTED** [review]

**Comment by S-NA** [comment]
@jtojnar changes applied, sorry for the oversight.

**Pull request merged**